examples/connext_dds/group_coherent_presentation/py/GroupCoherentExample_publisher.py

```diff
@@ -0,0 +1,155 @@
+    alarm_topic = dds.Topic(participant, "Alarm", Alarm)
+    heart_rate_topic = dds.Topic(participant, "HeartRate", HeartRate)
+    temperature_topic = dds.Topic(participant, "Temperature", Temperature)
+
+    publisher_qos = dds.QosProvider.default.publisher_qos
+    publisher = dds.Publisher(participant, publisher_qos)
+    writer_qos = dds.QosProvider.default.datawriter_qos
+
+    # Create one Writer per Topic using the Publisher and DataWriter QoS 
+    # specified in the xml file
+    alarm_writer = dds.DataWriter(publisher, alarm_topic, writer_qos)
+    heart_rate_writer = dds.DataWriter(publisher, heart_rate_topic, writer_qos)
+    temperature_writer = dds.DataWriter(
+        publisher,
+        temperature_topic,
+        writer_qos)
+
+    # Create the three datatypes and initialize the values
+    alarm_data = Alarm()
+    alarm_data.patient_id = 1
+    alarm_data.alarm_code = AlarmCode.PATIENT_OK
+
+    heart_rate_data = HeartRate()
+    heart_rate_data.patient_id = 1
+    heart_rate_data.beats_per_minute = 65
+
+    temperature_data = Temperature()
+    temperature_data.patient_id = 1
+    temperature_data.temperature = 98.6
+
+    # Below we will write a coherent set any time the patient's vitals are
+    # abnormal. Otherwise we will publish the patient's vital normally
+    for count in range(sample_count):
+        # Catch control c interrupt
+        try:
+            heart_rate_data.beats_per_minute = get_patient_heart_rate()
+            temperature_data.temperature = get_patient_temperature()
+            if heart_rate_data.beats_per_minute >= 100 or \
+                heart_rate_data.beats_per_minute <= 40 or \
+                temperature_data.temperature >= 100.0 or \
+                temperature_data.temperature <= 95.0:
+                # Sound an alarm. In this case, we want all of the
+                # patients vitals along with the alarm to be delivered as
+                # a single coherent set of data so that we can correlate
+                # the alarm with the set of vitals that triggered it
+                print("Patient's vitals are abnormal, raising alarm and"
+                      " sending coherent set")
+                with dds.CoherentSet(publisher):   # Start a coherent set
+                    heart_rate_writer.write(heart_rate_data)
+                    temperature_writer.write(temperature_data)
+
+                    alarm_data.alarm_code = AlarmCode.ABNORMAL_READING
+                    alarm_writer.write(alarm_data)
+                    # End of coherent set
+            else:
+                # No alarm necessary, publish the patient's vitals as normal
+                heart_rate_writer.write(heart_rate_data)
+                temperature_writer.write(temperature_data)
+                print("Writing patient's vitals...")
+            time.sleep(2)
+        except KeyboardInterrupt:
+            break
+
+    print('preparing to shut down...')
+    alarm_writer.close()
```

Comment:
no need to call close on all these entities. You can just call participant.close() -- and even that's optional, but it ensures that it happens before leaving the function.